# HG changeset patch
# User Maxim Dounin <mdounin@mdounin.ru>
# Date 1309188629 -14400
# Node ID f1ec9a0595de50832968569b85cee1073d1fa49b
# Parent  64ebc9b27e0ab7c1150a72f0a202fd14e0759ecb
Unbreak build with embedded perl and --with-openssl.
diff --git a/auto/lib/perl/make b/auto/lib/perl/make
--- a/auto/lib/perl/make
+++ b/auto/lib/perl/make
@@ -27,6 +27,7 @@ cat << END                              
 		&& NGX_PM_CFLAGS="\$(NGX_PM_CFLAGS) -g $NGX_CC_OPT"	\
 			NGX_PCRE=$PCRE					\
 			NGX_OBJS=$NGX_OBJS				\
+			NGX_OPENSSL=$OPENSSL				\
 		$NGX_PERL Makefile.PL					\
 			LIB=$NGX_PERL_MODULES				\
 			INSTALLSITEMAN3DIR=$NGX_PERL_MODULES_MAN
diff --git a/src/http/modules/perl/Makefile.PL b/src/http/modules/perl/Makefile.PL
--- a/src/http/modules/perl/Makefile.PL
+++ b/src/http/modules/perl/Makefile.PL
@@ -24,7 +24,11 @@ WriteMakefile(
                          "-I ../../../../../$ENV{NGX_OBJS} " .
                          ($ENV{NGX_PCRE} =~ /^(YES|NO)/ ? "" :
                              ($ENV{NGX_PCRE} =~ m#^/# ? "-I $ENV{NGX_PCRE} " :
-                                  "-I ../../../../../$ENV{NGX_PCRE} ")),
+                                  "-I ../../../../../$ENV{NGX_PCRE} ")) .
+                         ($ENV{NGX_OPENSSL} =~ /^(YES|NO)/ ? "" :
+                             ($ENV{NGX_OPENSSL} =~ m#^/# ?
+                                  "-I $ENV{NGX_OPENSSL}/.openssl/include " :
+                      "-I ../../../../../$ENV{NGX_OPENSSL}/.openssl/include ")),
 
     depend => {
         'nginx.c'     =>
